prov/usnic/src/usdf_msg.c

```diff
@@ -878,6 +883,7 @@ usdf_msg_handle_recv(struct usdf_domain *udp, struct usd_completion *comp)
 			rx_ptr += copylen;
 			rxlen -= copylen;
 			iov_resid -= copylen;
+			rqe->ms_resid -= copylen;
```

Comment:
Ditto here about `ms_resid` and using a local variable.

Also, in this `case RUDP_OP_LAST:`, I think you also should also add a small comment just after the `while` loop indicating that you would normally store back the updated values of `ms_cur_iov`, `ms_cur_ptr`, and `ms_iov_resid`, but you're trying to speed up the fastpath and these stores are not strictly necessary.